test/extended/kubevirt/services.go

```diff
@@ -0,0 +1,77 @@
+package kubevirt
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = Describe("[sig-kubevirt] services", func() {
+	oc := exutil.NewCLIWithPodSecurityLevel("ns-global", admissionapi.LevelBaseline)
+
+	InKubeVirtClusterContext(oc, func() {
+
+		mgmtFramework := e2e.NewDefaultFramework("mgmt-framework")
+		mgmtFramework.SkipNamespaceCreation = true
+
+		f1 := e2e.NewDefaultFramework("server-framework")
+		f1.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+
+		It("should allow connections to pods from infra cluster pod via NodePort across different infra nodes", func() {
+
+			// This tests connectivity from the infra cluster's pod network to a NodePort
+			// within a nested KubeVirt Hypershift guest cluster.
+			//
+			// This exercises the back half of the network flow used to pass ingress
+			// through the ingress cluster into the nested guest cluster.
+			//
+			// client pod (on infra cluster) -> guest node IP -> NodePort -> server pod (on guest cluster)
+			//
+			_, hcpNamespace, err := exutil.GetHypershiftManagementClusterConfigAndNamespace()
+			Expect(err).To(Succeed())
```

Comment:
`Expect(err).NotTo(HaveOccurred())`

you can do

```
err := functionThatReturnsAnError()
Expect(err).NotTo(HaveOccurred())
```

or

```
Expect(functionThatReturnsAnError()).To(Succeed())
```

but doing `Expect(err).To(Succeed())` makes it confusing to read. (What does it mean for an error to succeed?)

(This comment applies in a bunch of places.)